tests/unit/test_browse.py

```diff
@@ -298,39 +300,46 @@ def test_create_browse_imagery_with_mocks(
         )
 
     def test_convert_singleband_to_raster_without_colortable(self):
-        ds = MagicMock(DataArray)
-        ds.__getitem__.return_value = self.data
+        """Tests convert_gray_1band_to_raster."""
+
+        return_data = np.copy(self.data).astype('float64')
+        return_data[0][1] = np.nan
+        ds = DataArray(return_data).expand_dims('band')
 
         expected_raster = np.array(
             [
                 [
-                    [0, 104, 198, 255],
+                    [0, 0, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
                 ],
                 [
-                    [0, 104, 198, 255],
+                    [0, 0, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
                 ],
                 [
+                    [0, 0, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
                     [0, 104, 198, 255],
-                    [0, 104, 198, 255],
+                ],
+                [
+                    [255, 0, 255, 255],
+                    [255, 255, 255, 255],
+                    [255, 255, 255, 255],
+                    [255, 255, 255, 255],
                 ],
             ],
             dtype='uint8',
         )
-
         actual_raster = convert_singleband_to_raster(ds, None)
         assert_array_equal(expected_raster, actual_raster)
 
     def test_convert_singleband_to_raster_with_colormap(self):
-        ds = MagicMock(DataArray)
```

Comment:
What's the reason/advantage in not using MagicMock here anymore?

Reply:
Mostly it removes some of the intimate knowledge needed to use the mock. In this example I have to update the `__getitem__` return value, but in a different one I was changing `ds.to_numpy.return_value` by using the DataArray directly we don't have to worry about the internals.